src/batString.mlv

```diff
@@ -934,14 +953,15 @@ let numeric_compare s1 s2 =
   BatEnum.compare (fun g1 g2 ->
     let s1 = of_enum g1 in
     let s2 = of_enum g2 in
-    if BatChar.is_digit (unsafe_get s1 0) && BatChar.is_digit (unsafe_get s2 0) then
+    (* groups are never empty, so index 0 is always valid *)
+    if BatChar.is_digit (unsafe_get s1 0) &&
+       BatChar.is_digit (unsafe_get s2 0) then
```

Comment:
I don't trust the `Enum` module enough to ensure this property (or have you went to annotate all `Enum` functions involved here with static assertions to "prove" this as well?). You don't risk undefined behavior on trusting another module's interface specifications. I that you must use safe accessors here -- in any case, the overhead of Enum largely drowns out any bound access cost.

Reply:
Your arguments are convincing. I've reverted this change.